Remove tabs and unused local variables (DONE)

james-d-mitchell opened the pull request:
This PR removes all tabs and unused local variables from files starting "c" in the library. There were no tabs and no unused local variables in files starting "b" in the library, hence the lack of a PR for these.

## Text for release notes

None

fingolfin commented:
For the sake of efficiency, and to avoid having 20+ "convert tab commits" in the history and the `.git-blame-ignore-revs`, we could also have multiple of these "convert tabs" commits in a single PR; then we can review commit-by-commit to double check. And once all commits are "there" and reviewed, we could squash-merge.

This might also be easier for you compared to making many separate PRs?

That said, I am totally fine either way

james-d-mitchell commented:
I can do that if it's preferable, I thought you'd asked for a number of smaller PR's rather than one massive one, but perhaps I misunderstood.

james-d-mitchell commented:
I think it might take me a little while to do the rest of the files in the library, and it might be easier to merge this PR sooner rather than later (if it's acceptable) since otherwise the PR and `master` will diverge and become harder to merge.

fingolfin reviewed (commented):
Some minor oddities... no worries, we'll figure them out and then can merge (squash-merge?) this PR soonish. Need to sleep now, though

james-d-mitchell commented on lib/clashom.gi:
```diff
@@ -114,19 +114,19 @@ end);
-      bars,	# colour bars
+local clT,        # classes T
+      lcl,        # Length(clT)
+      clTR,        # classes under other group (autT,centralizer)
```
Good catch! I'll fix those just now, and continue :)

fingolfin commented on lib/clashom.gi:
```diff
@@ -147,22 +147,22 @@ local clT,	# classes T
-      stgen,	# local stabilizer generators
+      gen,        # generator
+      stab,        # stabilizer
+      stgen,        # local stabilizer generators
```
more comment indentation weirdness. Could it be that the script used to convert tabs to spaces blindly replaces each tab by 8 spaces, instead of aligning to the next multiple-of-8 column?

I am using the standard Unix util `expand` (following https://stackoverflow.com/a/11094620/928031 but adjusted to use tab width 8, i.e. removing the `-t 4`), and it seems to get these cases right.

james-d-mitchell replied:
Thanks @fingolfin, I am doing it by hand, so I'm to blame, I think I've fixed that now, I'll take a look at `expand`, sounds more efficient!

fingolfin replied:
Thanks James!

james-d-mitchell commented:
@fingolfin this is now complete, I've removed all unused local variables and tabs from `lib/*`.

ChrisJefferson reviewed (approved):
I had a quick look through and these all seem sensible. I looked closely at a few places where local variables had very suspicious names (like `PrintObj` or `Error`), but these variables were never used for anything, so removing them is safe (and good).

james-d-mitchell commented:
Thanks @ChrisJefferson, I was going to say: I'll squash the commits which remove tabs into one, and the same for the commits removing unused local variables, if this PR gets the green light.

fingolfin commented:
Thanks I'll have a look in a bit, too.

I am slightly vary about the files in hpcgap/lib/ getting out of sync. Perhaps there aren't too many tabs in them, though?

fingolfin commented:
Thanks I'll have a look in a bit, too.

I am slightly vary about the files in hpcgap/lib/ getting out of sync. Perhaps there aren't too many tabs in them, though?

james-d-mitchell commented:
Good point @fingolfin, there are apparently 167 lines with tabs in `hpcgap/lib` and a grand total of two local variables declared but not used.

fingolfin commented on lib/clashom.gi:
```diff
@@ -1645,41 +1645,29 @@ BindGlobal("LiftClassesEANonsolvGeneral",
+           cg,
            cNh,        # centralizer of <h> in <N>
-           C,  gens,   # preimage `Centralizer( cl )' under <hom>
+           gens,   # preimage `Centralizer( cl )' under <hom>
```
Removal of local variable caused a tiny misalignment here
```suggestion
           gens,       # preimage `Centralizer( cl )' under <hom>
```

fingolfin commented on lib/claspcgs.gi:
```diff
@@ -303,31 +303,31 @@ end );
+        field,      # field over which <N> is a vector space
+        h,          # preimage `cl.representative' under <hom>
+        cNh,        # centralizer of <h> in <N>
+        gens,   # preimage `Centralizer( cl )' under <hom>
```
```suggestion
        gens,       # preimage `Centralizer( cl )' under <hom>
```

fingolfin commented on lib/claspcgs.gi:
```diff
@@ -303,31 +303,31 @@ end );
+        n,  k,      # cf. Mecky--Neub\"user paper
+        cls,rep,pos,# set of classes with canonical representatives
+        j,
+        c,  ca,  i, # loop variables
+        S,                # orbit-stabilizer
+        ceve,#xponent vector
+        p,        # positions
+        Cgens,    # generators of C in N
+        next,blist, # orbit stabilizer algo
```
Not your fault, but while we are at it...
```suggestion
        c,  ca,  i, # loop variables
        S,          # orbit-stabilizer
        ceve,       # exponent vector
        p,          # positions
        Cgens,      # generators of C in N
        next,blist, # orbit stabilizer algo
```

fingolfin commented on lib/claspcgs.gi:
```diff
@@ -547,26 +547,26 @@ local  G,  home,  # the group and the home pcgs
        p,       # prime dividing $|G|$
        i,c,   # loop variables
-       opt,	# options
+       opt,     # options
```
```suggestion
       opt,      # options
```

fingolfin commented on lib/ctbllatt.gi:
```diff
@@ -1509,7 +1509,7 @@ InstallGlobalFunction( DnLattice, function( tbl, g1, y1 )
     i := 1;
     j := 0;
-    next[ 1	] := 1;
+    next[ 1     ] := 1;
```
```suggestion
    next[ 1 ] := 1;
```

fingolfin commented on lib/cyclotom.gi:
```diff
@@ -2005,24 +2005,24 @@ InstallGlobalFunction( DenominatorCyc, function( cyc )
-	Integers, GaussianIntegers,
-	Rationals, GaussianRationals,
-	Cyclotomics, fail
+        PositiveIntegers, NonnegativeIntegers,
```
I think in this file the tabs are due to me and were meant to be 4 wide. But not important

fingolfin reviewed (commented):
I've checke a-e, all good, some very minor suggestions for improvements. I'll look at the rest later.

fingolfin commented on lib/fpmon.gd:
```diff
@@ -1,7 +1,7 @@
 ##
 ##  This file is part of GAP, a system for computational discrete algebra.
-##  This file's authors include 			Isabel Araújo.
+##  This file's authors include                         Isabel Araújo.
```
```suggestion
##  This file's authors include Isabel Araújo.
```

fingolfin commented on lib/fpsemi.gi:
```diff
@@ -73,7 +73,7 @@ InstallMethod( \<,
     0,
     function( x1, x2 )
-			local S, RWS;
+                        local S, RWS;
```
maybe 4 (or 3?) wide tabs?

james-d-mitchell replied:
I've had another run through this file, super inconsistent indentation it seems!

fingolfin commented on lib/fpsemi.gi:
```diff
@@ -93,14 +93,14 @@ InstallMethod( \=,
-			RWS := ReducedConfluentRewritingSystem(S);
-			return ReducedForm(RWS, UnderlyingElement(x1)) =
-					ReducedForm(RWS, UnderlyingElement(x2));
+                        local S, RWS;
```
maybe 4 (or 3?) wide tabs?

fingolfin replied:
and more in this file

fingolfin replied:
could also be 2 again

fingolfin reviewed (commented):
Done a-f

james-d-mitchell commented on lib/factgrp.gi:
```diff
@@ -787,48 +787,48 @@ totalcnt, interrupt, u, nu, cor, zzz,bigperm,perm,badcores,max,i,hard;
-	# k"onnen)
+        # Abbruchkriterium: Bis kein Normalteiler, es sei denn, es ist N selber
+        # (das brauchen wir, um in einigen trivialen F"allen abbrechen zu
+        # k"onnen)
```
Indeed!

james-d-mitchell commented on lib/fastendo.gi:
```diff
@@ -41,7 +41,7 @@ function(dom, gmfam, trans)
   SetSource(tmap,dom);
   SetRange(tmap,dom);
-	SetIsEndoMapping(tmap, true);
+        SetIsEndoMapping(tmap, true);
```
I've had another run through that file too.

james-d-mitchell commented:
Ugh, and I just accidentally force pushed and removed the changes that I merged on GitHub directly, I'll fix those again, sorry. 

Also the latest commit in this PR removes some variables that are assigned but never used. I'm happy to put this in a separate PR if that's better, this change might be more problematic than the removal of unused local variables if the assignment of the variables has side effects. I've only removed those assigned but unused variables where I was fairly sure there'd be no side effects, in the other cases, I just remove the variable but left the right hand side of the assignment with a comment asking if the line is really required.

fingolfin commented on lib/grpfp.gi:
```diff
@@ -1931,15 +1918,14 @@ function ( G, H )
     freegens:=FreeGeneratorsOfFpGroup(Fam!.wholeGroup);
     # initialize the table for the intersection
-    rels := RelatorRepresentatives( RelatorsOfFpGroup( Fam!.wholeGroup ) );
+    RelatorRepresentatives( RelatorsOfFpGroup( Fam!.wholeGroup ) ); # TODO is this line required?
```
I believe it is safe to remove this

fingolfin commented on lib/grpfp.gi:
```diff
@@ -4883,7 +4863,7 @@ function(g)
   local i, rel,       # loop variable
         freegp,       # free group underlying g
-	id,	# identity of free group
+        id,        # identity of free group
```
```suggestion
        id,           # identity of free group
```

fingolfin commented on lib/grplatt.gi:
```diff
@@ -1708,7 +1704,7 @@ function (L)
             reps,              # transversal of <N> in <G>
-	    grp,	       # the group
+            grp,               # the group
             i,k,r;         # loop variables
```
```suggestion
            i,k,r;             # loop variables
```

fingolfin commented on lib/grplatt.gi:
```diff
@@ -1913,236 +1909,236 @@ local G,	# group
+                  for k in SmallGeneratingSet(G) do
+                    zim:=Coefficients(vsb,opr(Zero(vs),k));
+                    mat:=List(BasisVectors(vsb),x->
+            Concatenation(Coefficients(vsb,opr(x,k))-zim,[Zero(ocr.field)]));
```
```suggestion
                        Concatenation(Coefficients(vsb,opr(x,k))-zim,[Zero(ocr.field)]));
```

fingolfin commented on lib/grplatt.gi:
```diff
@@ -1913,236 +1909,236 @@ local G,	# group
+                        if not IsPcGroup(k) then
+                          k!.lattfpres:=ComplementFactorFpHom(
+                            ocr.factorfphom,l,M,N,k,ocr.generators,comp);
+            Assert(3,KernelOfMultiplicativeGeneralMapping(k!.lattfpres)=N);
```
```suggestion
                          Assert(3,KernelOfMultiplicativeGeneralMapping(k!.lattfpres)=N);
```

fingolfin commented on lib/grplatt.gi:
```diff
@@ -2427,7 +2423,7 @@ local uind,subs,incl,i,j,k,m,gens,t,c,p,conj,bas,basl,r;
   incl:=[];
   i:=1;
-  gens:=SmallGeneratingSet(U);
+  SmallGeneratingSet(U); # TODO is this line required?
```
Should be safe to remove

fingolfin commented on lib/grplatt.gi:
```diff
@@ -2814,54 +2808,54 @@ local fgi,inducedfactorautos,projs,psubs,info,n,l,nl,proj,emb,u,pos,
+          fi;
+          for s in subs do
+            for t in s[3] do # look over normals of subgroup
       #Print(t,"\n");
```
```suggestion
              #Print(t,"\n");
```

fingolfin commented on lib/grpnice.gi:
```diff
@@ -1026,7 +1026,7 @@ InstallMethod(NiceMonomorphism,
 #############################################################################
 ##
-#M  CanonicalNiceMonomorphism( <group> )	. .
+#M  CanonicalNiceMonomorphism( <group> )        . .
```
```suggestion
#M  CanonicalNiceMonomorphism( <group> )] . .
```

fingolfin commented on lib/grpnice.gi:
```diff
@@ -1040,7 +1040,7 @@ end);
 #############################################################################
 ##
-#M  SeedFaithfulAction( <group> )	. .
+#M  SeedFaithfulAction( <group> )        . .
```
```suggestion
#M  SeedFaithfulAction( <group> ) . .
```

fingolfin commented on lib/grppc.gi:
```diff
@@ -182,7 +182,7 @@ InstallMethod (InducedPcgsOp, "sift existing pcgs",
          od;
          if d> Length(pcgs) then
-         	Error ("Panic: Pcgs (G) does not seem to be a pcgs");
+                 Error ("Panic: Pcgs (G) does not seem to be a pcgs");
          else
```
```suggestion
            Error ("Panic: Pcgs (G) does not seem to be a pcgs");
         else
```

fingolfin commented on lib/grppc.gi:
```diff
@@ -440,7 +440,7 @@ function( elm, grp )
     if Length (pcgs) = Length (ppcgs) or not CanEasilyTestMembership (GroupOfPcgs(ppcgs)) then
         TryNextMethod();
-	fi;
+        fi;
     if elm in GroupOfPcgs (ppcgs) then
```
```suggestion
    fi;
    if elm in GroupOfPcgs (ppcgs) then
```

fingolfin commented on lib/grppc.gi:
```diff
@@ -1262,16 +1262,15 @@ end );
        Kp,Lp, # induced and modulo pcgs's
        KcapH,LcapH, # pcgs's of intersections with <H>
        N,   cent,   # elementary abelian factor, for affine action
        cls,  # classes in range/source of homomorphism
```
```suggestion
       cls,     # classes in range/source of homomorphism
```

fingolfin commented on lib/grppc.gi:
```diff
@@ -1295,7 +1294,7 @@ local  G,  home,  # the supergroup (of <H> and <U>), the home pcgs
 
   if IsPGroup( G )  then
-    p:=PrimePGroup(G);
+    PrimePGroup(G); # TODO is this line required?
```
can be removed

fingolfin commented on lib/grppc.gi:
```diff
@@ -1445,16 +1443,15 @@ InstallMethod(CentralizerModulo,"pcgs computable groups, for elm",
-      K,    L,     # members of <eas>
+      ea2,         # used for factor series
+      L,           # members of <eas>
       Kp,Lp,    # induced and modulo pcgs's
```
```suggestion
      Kp,Lp,       # induced and modulo pcgs's
```

fingolfin commented on lib/grppc.gi:
```diff
@@ -1483,30 +1480,30 @@ local G,	   # common parent
-	else
-	  eas:=fail; # useless
-	fi;
+        PrimePGroup(G);  # TODO is this line required?
```
can be removed

fingolfin commented on lib/grppcaut.gi:
```diff
@@ -1603,49 +1603,49 @@ end);
+        fi;
+
+        if IsPGroup(G) then
+                return fail;        # p-groups should be handled elsewhere
```
Maybe this function once again uses tab width 4 or so?

fingolfin reviewed (commented):
Done until and including grppccom.gi

A bunch of minor remarks; many are about formatting issues that were present before

fingolfin commented on lib/grppcnrm.gi:
```diff
@@ -636,10 +636,10 @@ PCGS_NORMALIZER_COBOUNDS := function( home, pcgs, nis, u1, u2 )
-    	for j  in [ 1 .. Length(mats) ]  do
-    	    Append( l[i], t[i]-mats[j][i] );
-    	od;
+            l[i] := [];
+            for j  in [ 1 .. Length(mats) ]  do
+                Append( l[i], t[i]-mats[j][i] );
+            od;
```
```suggestion
        l[i] := [];
        for j  in [ 1 .. Length(mats) ]  do
            Append( l[i], t[i]-mats[j][i] );
        od;
```

fingolfin reviewed (commented):
Done a-l

fingolfin commented on lib/oprtpcgs.gi:
```diff
@@ -14,13 +14,13 @@
-	i, ii, j, k;     # loop variables
+        len,             # lengths of orbit before each extension
+        d,               # dictionary
+        S, rel,   # stabilizer and induced pcgs
```
```suggestion
        S, rel,          # stabilizer and induced pcgs
```

fingolfin reviewed (commented):
Completed a-o

ChrisJefferson commented:
If this PR is squashed (not saying it has to be), could it be put into at least 3 commits?

* Whitespace,
* Remove variables which are never used,
* Remove variables which were assigned, but then never used.

james-d-mitchell commented:
@ChrisJefferson yes, I'll do that once everything is ok

james-d-mitchell commented on lib/galois.gi:
```diff
@@ -270,10 +268,10 @@ end);
   Info(InfoGalois,3,"ProductRootsPol ",m);
   fam:=CoefficientsFamily(FamilyObj(f));
-  ch:=Characteristic(fam);
+  Characteristic(fam); # TODO is this line required?
```
Removed!

james-d-mitchell commented on lib/ghomfp.gi:
```diff
@@ -1138,21 +1133,21 @@ InstallMethod(MaximalAbelianQuotient,
     generatorsLimit:=Length(GeneratorsOfGroup(Parent(U)))*LogInt(IndexInWholeGroup(U),2),
     cheap);
-  m:=MaximalAbelianQuotient(Image(phi));
+  MaximalAbelianQuotient(Image(phi)); # TODO is this line required?
```
Done.

james-d-mitchell commented:
Thanks for the comments @fingolfin, I imagine it was as interesting to look at the changes in these files as it was to make them :) I think I've addressed all of your comments now, please let me know when you're done, so I can squash the commits according to what @ChrisJefferson suggested.

fingolfin commented on lib/randiso.gi:
```diff
@@ -563,16 +563,16 @@ InstallGlobalFunction( RandomIsomorphismTest, function( list, n )
         od;
 
-		# just for information
+                # just for information
```
Something not right here

fingolfin commented on lib/relation.gi:
```diff
@@ -122,7 +122,7 @@
 ##
 DeclareRepresentation("IsBinaryRelationDefaultRep",
-	IsAttributeStoringRep ,[]);
+        IsAttributeStoringRep ,[]);
```
```suggestion
        IsAttributeStoringRep, []);
```

fingolfin commented on lib/relation.gi:
```diff
@@ -131,7 +131,7 @@ DeclareRepresentation("IsBinaryRelationDefaultRep",
 ##
 DeclareRepresentation("IsBinaryRelationOnPointsRep",
-	IsAttributeStoringRep ,[]);
+        IsAttributeStoringRep ,[]);
```
```suggestion
        IsAttributeStoringRep, []);
```

fingolfin commented on lib/relation.gi:
```diff
@@ -140,7 +140,7 @@ DeclareRepresentation("IsBinaryRelationOnPointsRep",
 ##
 DeclareRepresentation("IsEquivalenceRelationDefaultRep",
-	IsAttributeStoringRep ,[]);
+        IsAttributeStoringRep ,[]);
```
```suggestion
        IsAttributeStoringRep, []);
```

fingolfin commented on lib/relation.gi:
```diff
@@ -696,110 +696,7 @@ InstallMethod(StronglyConnectedComponents, "for general binary relations",
-
-        ## Recursive version of a depth first visit
-        ##
-        DFSVisit := function(u)
```
So there is a bunch of dead code being removed here, and I am a bit torn about that: on the one hand, it is dead, and it's good we recognized that. And it'll likely always stay dead and so we can remove it... On the other hand, I really wonder what the story here is. Was this code simply superseded by something better? Or was there a bug and it was disabled? Or what else is going on? <ight try to dig into `git blame` to find out who wrote this...

james-d-mitchell replied:
I think it's 

> Was this code simply superseded by something better? 

I'll double check and write some more in a moment.

james-d-mitchell replied:
I think it's completely safe to remove these lines of code, the method is to compute the strongly connected components of a binary relation, it looks like the old (removed) code did this using a DFS, and the newer code uses the kernel function `STRONGLY_CONNECTED_COMPONENTS_DIGRAPH`.

fingolfin commented on lib/relation.gi:
```diff
@@ -1413,36 +1287,36 @@ InstallMethod(PrintObj, "for a binary relation on  n points", true,
-		Error("Input does not describe a partition");
-	fi;
+        if not  (IsSubset(X, AsSortedList(Concatenation(subsX)))) then
+                Error("Input does not describe a partition");
```
Weird indentation here, makes me wonder if this function should once again use less wide tabs... Except that there are a few bits and pieces (like the comment just above this place here) which are indented with spaces and fit with 8-wide tabs... But this might be an artifact of later edits.. Urgh

james-d-mitchell replied:
Agreed it's weird, this file is one of the more inconsistent ones, I could re-indent the entire file, or just leave it?

fingolfin commented on lib/relation.gi:
```diff
@@ -1967,7 +1841,7 @@ InstallMethod( PrintObj, "for an equivalence relation", true,
 #M  EquivalenceClasses( <E> )
 ##
-##	Wraparound function which calls the two-argument method
+##      Wraparound function which calls the two-argument method
```
```suggestion
##  Wraparound function which calls the two-argument method
```

fingolfin commented on lib/ring.gi:
```diff
@@ -1203,7 +1203,7 @@ InstallGlobalFunction( Gcd, function ( arg )
      IsIdenticalObj(FamilyObj(arg[1]),FamilyObj(arg[2]))
       then # quick dispatch for two ring elements. There is a
-	   # fallback method that still supplies the ring if nothing special
+           # fallback method that still supplies the ring if nothing special
```
```suggestion
       # fallback method that still supplies the ring if nothing special
```

fingolfin commented on lib/rwssmg.gi:
```diff
@@ -36,11 +36,11 @@ function(kbrws)
-	elif IsElementOfFpMonoidFamily(FamilyForRewritingSystem(kbrws)) then
-		SetIsBuiltFromMonoid(rws,true);
-	fi;
+        if IsElementOfFpSemigroupFamily(FamilyForRewritingSystem(kbrws)) then
```
I think this needs 2-wide tabs (possibly the whole file)

james-d-mitchell replied:
Fixed

fingolfin commented on lib/schur.gi:
```diff
@@ -211,7 +211,7 @@ local s,pcgs,n,iso,H,l,cov,der,pco,ng,gens,imgs,ran,zer,i,j,e,a,
   # (use Burnside's basis theorem)
 
-  der:=DerivedSubgroup(cov);
+  DerivedSubgroup(cov); # TODO is this line required?
```
Can go

james-d-mitchell replied:
Removed

fingolfin commented on lib/semicong.gi:
```diff
@@ -1,7 +1,7 @@
 ##
 ##  This file is part of GAP, a system for computational discrete algebra.
-##  This file's authors include 	               Andrew Solomon.
+##  This file's authors include                            Andrew Solomon.
```
```suggestion
##  This file's authors include Andrew Solomon.
```

fingolfin commented on lib/stbcbckt.gi:
```diff
@@ -214,10 +214,9 @@ InstallGlobalFunction( StratMeetPartition, function( arg )
             pnt,    # fixpoint to be processed
             cellsP, #\
             blist,  #  >see explanation below
```
```suggestion
            blist,  #/ see explanation below
```

fingolfin commented on lib/stbcbckt.gi:
```diff
@@ -1138,7 +1136,6 @@ InstallGlobalFunction( PartitionBacktrack,
         local  undoto,   # number of cells of <P> wanted after undoing
                oldprm,   #\
-               oldrgt,   #  > old values of <image>
                oldprm2,  #/
```
```suggestion
               oldprm2,  #/  old values of <image>
```

fingolfin commented on lib/stbcbckt.gi:
```diff
@@ -1263,10 +1260,10 @@ InstallGlobalFunction( PartitionBacktrack,
-	fi;
+        if d=1 and ForAll(GeneratorsOfGroup(G),x->a^x=a) then
+          orb[d][a]:=true; # ensure a is a possible image (can happen if
+                          # acting on permutations with more points)
```
```suggestion
                           # acting on permutations with more points)
```

fingolfin commented on lib/streams.gi:
```diff
@@ -133,7 +133,7 @@ end);
 #############################################################################
 ##
-#M  Read( <input-stream> )	. . . . . . . . . .  read stream as GAP input
+#M  Read( <input-stream> )      . . . . . . . . . .  read stream as GAP input
```
```suggestion
#M  Read( <input-stream> )  . . . . . . . . . . . .  read stream as GAP input
```

fingolfin commented on lib/tietze.gi:
```diff
@@ -656,7 +654,7 @@ InstallGlobalFunction( PresentationRegularPermutationGroupNC, function ( G )
                             app[10] := g;
                             app[11] := c;
-                            n := MakeConsequences( app );
+                            MakeConsequences( app ); # TODO is this line required?
```
No idea :-/

fingolfin commented on lib/twocohom.gi:
```diff
@@ -689,7 +689,7 @@ function( G, M )
     cb := TwoCoboundariesSQ( C, G, M );
     cb := SubspaceNC( co, cb );
-    pr := FpGroupPcGroupSQ( G );
+    FpGroupPcGroupSQ( G ); # TODO is this line required?
```
I don't think so

james-d-mitchell replied:
Removed

fingolfin reviewed (commented):
covered a-z now

fingolfin commented:
@james-d-mitchell how shall we proceed to get this merged? There are still comments by me (which I could apply, or not, or you; I am fine either way, just don't want to cause accidents).

Also, the linter CI actually fails?

fingolfin commented on .gaplint.yml:
```diff
@@ -1,5 +1,4 @@
 disable:
-- W000
```
I think these `.gaplint.yml` are not possible while there are still files in `hpcgap/lib/` which are not de-tabbed

fingolfin replied:
I've restarted the gaplint job. If it finishes, please squash as described.

The `testexpect` failure is indeed separate, see issue #5309

james-d-mitchell replied:
Will squash now thanks!

james-d-mitchell replied:
Squashed, hopefully correctly.

james-d-mitchell commented on lib/tietze.gi:
```diff
@@ -656,7 +654,7 @@ InstallGlobalFunction( PresentationRegularPermutationGroupNC, function ( G )
                             app[10] := g;
                             app[11] := c;
-                            n := MakeConsequences( app );
+                            MakeConsequences( app ); # TODO is this line required?
```
Thanks @fingolfin !
